UCS SAN Connectivity Policy module and integration test

dsoper2 opened the pull request:
##### SUMMARY
<!--- Describe the change, including rationale and design decisions -->
SAN Connectivity Policies and integration test
<!---
If you are fixing an existing issue, please include "Fixes #nnn" in your
commit message and your description; but you should still explain what
the change does.
-->

##### ISSUE TYPE
<!--- Pick one below and delete the rest: -->
 - New Module Pull Request


##### COMPONENT NAME
<!--- Name of the module, plugin, module or task -->
ucs_san_connectivity
##### ANSIBLE VERSION
<!--- Paste verbatim output from "ansible --version" between quotes below -->
```
ansible 2.5.0 (ucsm 1bd09fa1a9) last updated 2017/12/18 16:44:38 (GMT -500)
  config file = None
  configured module search path = [u'/Users/dsoper/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/dsoper/Documents/ansible/lib/ansible
  executable location = /Users/dsoper/Documents/ansible/bin/ansible
  python version = 2.7.12 (default, Oct 11 2016, 05:20:59) [GCC 4.2.1 Compatible Apple LLVM 8.0.0 (clang-800.0.38)]
```


##### ADDITIONAL INFORMATION
<!---
Include additional information to help people understand the change here.
For bugs that don't have a linked bug report, a step-by-step reproduction
of the problem is helpful.
  -->

<!--- Paste verbatim command output below, e.g. before and after your change -->
```

```

Commit:
vhba template and integration test

Commit:
SAN Connectivity Policies and integration test

Commit:
VSANs and integration tests

Commit:
WWNN/WWPN Pools and integration test

Commit:
SAN Connectivity Policies only

dagwieers commented on lib/ansible/modules/remote_management/ucs/ucs_san_connectivity.py:
```diff
@@ -0,0 +1,219 @@
+    description:
+    - Name of the SAN Connectivity Policy
+    - If specifying a single SAN Connectivity Policy, name is required
+  descr:
```
For ACI we opted to use `description` as default, and made an alias for `descr`. Since otherwise it looks a bit awkward (and core devs have questioned it too). But it's up to you.

dsoper2 replied:
Spelled out in the latest commit

Commit:
option description spelled out (aliased to descr)
removed docs for list of dictionaries syntax
follow ACI indent style
state absent only requires dn name match (not full prop match)

dagwieers commented on lib/ansible/modules/remote_management/ucs/ucs_san_connectivity.py:
```diff
@@ -0,0 +1,235 @@
+                    child_dn = dn + '/fc-node'
+                    mo_1 = ucs.login_handle.query_dn(child_dn)
+                    if mo_1:
+                        kwargs = {}
```
You could create the dict with the entry in one go:
```python
kwargs = dict(ident_pool_name=san_connectivity['wwn_pool'])
```
Same below.

dagwieers reviewed (approved):
LGTM

dagwieers commented:
I guess the descriptions could be improved, since they don't add any value here. But I am going to merge this, and then you can update the documentation later.
